hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -706,10 +717,11 @@ ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
   //  Start with relations and try to split
   for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
   {
-    for (ChangesetInfo::iterator it = changeset->begin(ElementType::Relation, (ChangesetType)current_type);
-         it != changeset->end(ElementType::Relation, (ChangesetType)current_type); ++it)
+    //  Because we are modifying the changeset, get the first element in the set and iterate
+    while (changeset->size(ElementType::Relation, (ChangesetType)current_type) > 0)
```

Comment:
The `for` loop on an iterator that points to an object that is modified causes issues.  This `while` loop checks the first element until there aren't any more elements to check.